dependency_cache.go

```diff
@@ -177,14 +178,20 @@ func (d *DependencyCache) Artifact(dependency BuildpackDependency, mods ...Reque
 		}
 	}
 
+	urlP, err := url.Parse(uri)
+	if err != nil {
+		d.Logger.Debugf("URI format invalid\n%w", err)
+		return nil, fmt.Errorf("unable to parse URI. see DEBUG log level")
+	}
+
 	if dependency.SHA256 == "" {
 		d.Logger.Headerf("%s Dependency has no SHA256. Skipping cache.",
 			color.New(color.FgYellow, color.Bold).Sprint("Warning:"))
 
-		d.Logger.Bodyf("%s from %s", color.YellowString("Downloading"), uri)
-		artifact = filepath.Join(d.DownloadPath, filepath.Base(uri))
-		if err := d.download(uri, artifact, mods...); err != nil {
-			return nil, fmt.Errorf("unable to download %s\n%w", uri, err)
+		d.Logger.Bodyf("%s from %s", color.YellowString("Downloading"), urlP.Redacted())
+		artifact = filepath.Join(d.DownloadPath, filepath.Base(urlP.Path))
```

Comment:
```suggestion
		artifact = filepath.Join(d.DownloadPath, filepath.Base(uri))
```

I'd like to change this back to use the base of the URI. There is technically a difference if you have query parameters on the URI and I'm not sure if it would impact anyone, but I'd rather not chance it. I don't think this affects the redaction of values, so I think this should be OK, but please let me know if you have concerns.